tests/Unit/ImageTest.php

```diff
@@ -109,16 +63,33 @@ public function ensureProcessedFilesAreAvailable(Version $version)
     #[Depends('ensureProcessedFilesAreAvailable')]
     public function ensureUnprocessedFilesAreNotAvailable(Version $version)
     {
-        $version->update(['processed' => 0]);
-        $getDerivativeResponse = $this->get(route('getImageDerivative', [$this->user->name, $version->Media]));
+        $version->Media->Versions->each->update(['processed' => 0]);
 
+        $getDerivativeResponse = $this->get(route('getImageDerivative', [$this->user->name, $version->Media]));
         $getDerivativeResponse->assertNotFound();
     }
 
+    #[Test]
+    #[Depends('ensureVersionCanBeSet')]
+    public function ensureDeletingAVersionDoesNotDeleteOriginalFileIfAVersionPointingToThisFileStillExists(Version $version)
+    {
+        $originalVersion = $version->Media->Versions()->where('filename', $version->filename)->oldest()->first();
+
+        $this->assertNotEquals($version->id, $originalVersion->id);
+
+        $this->originalsDisk->assertExists($version->originalFilePath());
+        $this->originalsDisk->assertExists($originalVersion->originalFilePath());
+        $this->assertEquals($this->originalsDisk->path($version->originalFilePath()), $this->originalsDisk->path($originalVersion->originalFilePath()));
+
+        $version->delete();
```

Comment:
maybe also assert that the deletion was successful

Reply:
done.